modules/packages/NPBRand.chpl

```diff
@@ -0,0 +1,508 @@
+      return (randlc(x), randlc(x)):complex;
+    else
+      if resultType == imag then
+        //
+        // BLC: I thought that casting real to imag did this automatically?
+        //
+        return _r2i(randlc(x));
+      else
+        return randlc(x);
+    }
+
+  //
+  // Return a value for the cursor so that the next call to randlc will
+  // return the same value as the nth call to randlc
+  //
+  private proc randlc_skipto(seed: int(64), in n: integral): real {
+    var cursor = seed:real;
+    n -= 1;
+    var t = arand;
+    arand;
+    while (n != 0) {
+      const i = n / 2;
+      if (2 * i != n) then
+        randlc(cursor, t);
+      if i == 0 then
+        break;
+      else
+        n = i;
+      randlc(t, t);
+      n = i;
+    }
+    return cursor;
+  }
+
+  //
+  // iterate over outer ranges in tuple of ranges
+  //
+  private iter outer(ranges, param dim: int = 0) {
+    if dim + 2 == ranges.size {
+      foreach i in ranges(dim) do
+        yield (i,);
+    } else if dim + 2 < ranges.size {
+      foreach i in ranges(dim) do
+        foreach j in outer(ranges, dim+1) do
+          yield (i, (...j));
+    } else {
+      yield 0; // 1D case is a noop
+    }
+  }
+
+  //
+  // RandomStream iterator implementation
+  //
+  @chpldoc.nodoc
+  iter NPBRandomPrivate_iterate(type resultType, D: domain, seed: int(64),
+                        start: int(64)) {
+    var cursor = randlc_skipto(seed, start);
+    for i in D do
+      yield randlc(resultType, cursor);
+  }
+
+  @chpldoc.nodoc
+  iter NPBRandomPrivate_iterate(type resultType, D: domain, seed: int(64),
+                        start: int(64), param tag: iterKind)
+        where tag == iterKind.leader {
+    // forward to the domain D's iterator
+    for block in D.these(tag=iterKind.leader) do
+      yield block;
+  }
+
+  @chpldoc.nodoc
+  iter NPBRandomPrivate_iterate(type resultType, D: domain, seed: int(64),
+                start: int(64), param tag: iterKind, followThis)
+        where tag == iterKind.follower {
+    use DSIUtil;
+    param multiplier = if resultType == complex then 2 else 1;
+    const ZD = computeZeroBasedDomain(D);
+    const innerRange = followThis(ZD.rank-1);
+    var cursor: real;
+    for outer in outer(followThis) {
+      var myStart = start;
+      if ZD.rank > 1 then
+        myStart += multiplier * ZD.indexOrder(((...outer), innerRange.lowBound)).safeCast(int(64));
+      else
+        myStart += multiplier * ZD.indexOrder(innerRange.lowBound).safeCast(int(64));
+      if innerRange.hasUnitStride() {
+        cursor = randlc_skipto(seed, myStart);
+        for i in innerRange do
+          yield randlc(resultType, cursor);
+      } else {
+        myStart -= innerRange.lowBound.safeCast(int(64));
+        for i in innerRange {
+          cursor = randlc_skipto(seed, myStart + i.safeCast(int(64)) * multiplier);
+          yield randlc(resultType, cursor);
+        }
+      }
+    }
+  }
+} // close module NPBRandom
```

Comment:
I did not look closely at this file as I assume it is pretty much all code movement without functionality changes.

IMO this is evidence that the module should be called NPBRandom.